test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -0,0 +1,1031 @@
+        });
+      });
+      itp('does not call transformUsername while loading security image', function () {
+        return setup({ features: { securityImage: true }, transformUsername: transformUsername })
+        .then(function (test) {
+          spyOn(test.router.settings, 'transformUsername');
+          test.setNextResponse(resSecurityImage);
+          test.form.setUsername('testuser@clouditude.net');
+          return waitForBeaconChange(test);
+        })
+        .then(function (test) {
+          expect(test.router.settings.transformUsername.calls.count()).toBe(0);
+          expect($.ajax.calls.count()).toBe(1);
+          expect($.ajax.calls.argsFor(0)[0]).toEqual({
+            url: 'https://foo.com/login/getimage?username=testuser@clouditude.net',
+            type: 'get',
+            dataType: undefined,
+            data: undefined,
+            success: undefined
+          });
+        });
+      });
+      itp('changs the suffix of the username', function () {
+        return setupWithTransformUsername().then(function (test) {
+          test.form.setUsername('testuser@clouditude.net');
+          test.setNextWebfingerResponse(resSuccess);
+          test.form.submit();
+          return tick(test);
+        }).then(function (test) {
+          expect(test.ac.webfinger).toHaveBeenCalledWith({
+            resource: 'acct:testuser@example.com',
+            requestContext: undefined
+          });
+        });
+      });
+      itp('does not change the suffix of the username if "IDP_DISCOVERY" operation is not handled', function () {
+        return setupWithTransformUsernameOnUnlock().then(function (test) {
+          test.form.setUsername('testuser@clouditude.net');
+          test.setNextWebfingerResponse(resSuccess);
+          test.form.submit();
+          return tick(test);
+        }).then(function (test) {
+          expect(test.ac.webfinger).toHaveBeenCalledWith({
+            resource: 'acct:testuser@clouditude.net',
+            requestContext: undefined
+          });
+        });
+      });
+    });
+
+    Expect.describe('events', function () {
+
+      Expect.describe('beacon loading', function () {
+        itp('shows beacon-loading animation when authClient webfinger is called', function () {
+          return setup({ features: { securityImage: true }})
+          .then(function (test) {
+            test.securityBeacon = test.router.header.currentBeacon.$el;
+            test.setNextResponse(resSecurityImage);
+            test.form.setUsername('testuser@clouditude.net');
+            return waitForBeaconChange(test);
+          })
+          .then(function (test) {
+            spyOn(test.securityBeacon, 'toggleClass');
+            test.setNextWebfingerResponse(resSuccess);
+            test.form.submit();
+            return tick(test);
+          })
+          .then(function (test) {
+            return tick(test);
```

Comment:
this is an example where ```tick``` is not the best, since we are calling it twice. If we can call a different 'waitFor` function it will be much better